docs/manual/java/guide/services/AdditionalRouters.md

```diff
@@ -0,0 +1,132 @@
+
+There are two variants for `additionalRouters`, one that receives a `Class<play.api.routing.Router>` and one that receives a `play.api.routing.Router`.
+
+The first one should be used when your Router has some dependencies and needs to get them injected  by Lagom's runtime DI infrastructure (Guice). In this case, you pass the class and Guice will initialize it with the right dependencies.
+
+
+```java
+public class HelloWorldModule extends AbstractModule implements ServiceGuiceSupport {
+    @Override
+    protected void configure() {
+        bindService(
+                HelloService.class, HelloServiceImpl.class,
+                additionalRouter(SomeOtherPlayRouter.class)
+        );
+    }
+}
+```
+
+The second variant should be used be used when the Router does not have any other dependencies and therefore can be immediately passed as an instance.
+
+```java
+public class HelloWorldModule extends AbstractModule implements ServiceGuiceSupport {
+    @Override
+    protected void configure() {
+        bindService(
+                HelloService.class, HelloServiceImpl.class,
+                additionalRouter(new SomeOtherPlayRouter())
+        );
+    }
+}
+```
+
+
+
+## File Upload Example
+
+The following example shows how you can add a file upload endpoint to an existing Lagom Service. 
+
+The example is based on [JavaRoutingDsl](https://www.playframework.com/documentation/2.7.x/JavaRoutingDsl) that allows you to build a Play Router programmatically. It adds an extra path (`/api/files`) that receives POST calls for multipart-form data. 
+
+```java
+class FileUploadRouter implements SimpleRouter {
+
+    private final Router delegate;
+
+    @Inject
+    public FileUploadRouter(RoutingDsl routingDsl) {
+        this.delegate = routingDsl
+                .POST("/api/files")
+                .routingTo( __ -> {
+                    // for the sake of simplicity, this implementation
+                    // only returns a short message for each incoming request.
+                    return ok("File(s) uploaded");
+                })
+                .build().asScala();
+    }
+
+    @Override
+    public PartialFunction<RequestHeader, Handler> routes() {
+        return delegate.routes();
+    }
+}
+```
+
+In your Guice module, you append the additional router when binding your Lagom service.
+
+```java
+public class HelloWorldModule extends AbstractModule implements ServiceGuiceSupport {
+    @Override
+    protected void configure() {
+        bindService(
+                HelloWorldService.class, HelloWorldServiceImpl.class,
+                additionalRouter(FileUploadRouter.class)
+        );
+    }
+}
+
+```
+
+The path `/api/files` will now be exposed on your Lagom service:
```

Comment:
```suggestion
The path `/api/files` will now be available on your Lagom service:
```

I think saying `will be exposed` is confusing since it's not explicitly `exposed` via the `Service.Descriptor`.